js/battle.js

```diff
@@ -6104,19 +6104,24 @@ var Battle = (function () {
 				}
 				if (isSwitch && pokemon == this.p1.lastPokemon && !this.p1.active[slot]) continue;
 				if ((searchid && pokemon.searchid === searchid) || // exact match
-					(!pokemon.searchid && pokemon.checkDetails(details)) || // switch-in matches Team Preview entry
 					(!searchid && pokemon.ident === pokemonid)) { // name matched, good enough
 					if (!pokemon.searchid && createIfNotFound) {
 						pokemon.name = name;
 						pokemon.searchid = searchid;
 						pokemon.ident = pokemonid;
-						if (pokemon.needsReplace) {
-							pokemon = this.p1.newPokemon(this.parseDetails(name, pokemonid, details), i);
-						}
 					}
 					if (slot >= 0) pokemon.slot = slot;
 					return pokemon;
 				}
+				if (!pokemon.searchid && (pokemon.checkDetails(details) || // switch-in matches Team Preview entry
+					(!pokemon.shiny && pokemon.checkDetails(details.replace(', shiny', ''))))) { // switch-in matches non-shiny Team Preview entry
```

Comment:
You don't need `!pokemon.shiny`; that's sufficiently implicated by `!pokemon.searchid`